cmd/block-generator/runner/run.go

```diff
@@ -0,0 +1,260 @@
+		// Skip this one
+		if metric == "genesis" {
+			continue
+		}
+		str := fmt.Sprintf("transaction_%s_total:%d\n", metric, entry.GenerationCount)
+		if _, err := report.WriteString(str); err != nil {
+			return fmt.Errorf("unable to write transaction_count metric: %w", err)
+		}
+	}
+
+	// Record a rate from one of the first data points.
+	if len(collector.Data) > 5 {
+		if err := recordDataToFile(collector.Data[2], "starting_block_import_duration_average_seconds", report); err != nil {
+			return err
+		}
+	}
+
+	// Also record the final one.
+	if err := recordDataToFile(collector.Data[len(collector.Data)-1], "final_import_duration_average_second", report); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// startGenerator starts the generator server.
+func startGenerator(configFile string, addr string) func() error {
+	// Start generator.
+	server, done := generator.StartServer(configFile, addr)
+
+	return func() error {
+		if err := server.Shutdown(context.Background()); err != nil {
+			return fmt.Errorf("failed during generator graceful shutdown: %w", err)
+		}
+
+		// Wait for graceful shutdown or crash.
+		select {
+		case <-done:
+			// continue
+			return nil
+		case <-time.After(10 * time.Second):
+			return fmt.Errorf("failed to gracefully shutdown generator")
+		}
+	}
+}
+
+// startIndexer resets the postgres database and executes the indexer binary. It performs some simple verification to
+// ensure that the service has started properly.
+func startIndexer(indexerBinary string, algodNet string, indexerNet string, postgresConnectionString string) (func() error, error) {
+	{
+		db, err := sql.Open("postgres", postgresConnectionString)
+		if err != nil {
+			return nil, fmt.Errorf("postgres connection string did not work: %w", err)
+		}
+		db.Exec(`DROP SCHEMA public CASCADE; CREATE SCHEMA public;`)
+		db.Close()
+	}
+
+	time.Sleep(250 * time.Millisecond)
+
+	cmd := exec.Command(
+		indexerBinary,
+		"daemon",
+		"--algod-net", algodNet,
+		"--algod-token", "secure-token-here",
+		"--metrics-mode", "VERBOSE",
+		"--postgres", postgresConnectionString,
+		"--server", indexerNet)
+
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	if err := cmd.Start(); err != nil {
+		return nil, fmt.Errorf("failure calling Start(): %w", err)
+	}
+
+	// Ensure that the health endpoint can be queried.
+	// The service should start very quickly because the DB is empty.
+	time.Sleep(5 * time.Second)
+	resp, err := http.Get(fmt.Sprintf("http://%s/health", indexerNet))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "stdout:\n%s\n", stdout.String())
+		fmt.Fprintf(os.Stderr, "stderr:\n%s\n", stderr.String())
+		return nil, fmt.Errorf("the process failed to start properly, health endpoint query failed")
+	}
+	resp.Body.Close()
+
+	return func() error {
+		if err := cmd.Process.Kill(); err != nil {
+			return fmt.Errorf("failed to kill indexer process: %w", err)
+		}
+
+		// Clear postgres DB
```

Comment:
?

Reply:
I moved this to the beginning. Removed comment.